File: awesomedialoglib/src/main/java/com/awesomedialog/blennersilva/awesomedialoglibrary/AwesomeSuccessDialog.java
package com.awesomedialog.blennersilva.awesomedialoglibrary;

import android.content.Context;
import android.graphics.PorterDuff;
import android.support.annotation.Nullable;
import android.support.v4.content.ContextCompat;
import android.view.View;
import android.widget.Button;
import android.widget.RelativeLayout;

import com.awesomedialog.blennersilva.awesomedialoglibrary.interfaces.Closure;

/**
 * Created by blennersilva on 23/08/17.
 */

public class AwesomeSuccessDialog extends AwesomeDialogBuilder<AwesomeSuccessDialog> {

    private Button positiveButton;
    private Button negativeButton;
    private Button doneButton;
    private RelativeLayout dialogBody;

    public AwesomeSuccessDialog(Context context) {
        super(context);

        setColoredCircle(R.color.dialogSuccessBackgroundColor);
        setDialogIconAndColor(R.drawable.ic_success, R.color.white);
        setNegativeButtonbackgroundColor(R.color.dialogSuccessBackgroundColor);
        setPositiveButtonbackgroundColor(R.color.dialogSuccessBackgroundColor);
        setDoneButtonbackgroundColor(R.color.dialogSuccessBackgroundColor);
    }

    {
        positiveButton = findView(R.id.btDialogYes);
        negativeButton = findView(R.id.btDialogNo);
        doneButton = findView(R.id.btDialogDone);
        dialogBody = findView(R.id.dialog_body);
    }

    public AwesomeSuccessDialog setDialogBodyBackgroundColor(int color){
        if (dialogBody != null) {
            dialogBody.getBackground().setColorFilter(ContextCompat.getColor(getContext(), color), PorterDuff.Mode.SRC_IN);
        }

        return this;
    }

    public AwesomeSuccessDialog setPositiveButtonClick(@Nullable final Closure selectedYes) {
        positiveButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (selectedYes != null) {
                    selectedYes.exec();
                }

                hide();
            }
        });

        return this;
    }

    public AwesomeSuccessDialog setNegativeButtonClick(@Nullable final Closure selectedNo) {
        negativeButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (selectedNo != null) {
                    selectedNo.exec();
                }

                hide();
            }
        });

        return this;
    }

    public AwesomeSuccessDialog setDoneButtonClick(@Nullable final Closure selectedDone) {
        doneButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (selectedDone != null) {
                    selectedDone.exec();
                }

                hide();
            }
        });

        return this;
    }

    public AwesomeSuccessDialog showPositiveButton(boolean show) {
        if (positiveButton != null) {
            positiveButton.setVisibility(View.VISIBLE);
        }

        return this;
    }

    public AwesomeSuccessDialog showNegativeButton(boolean show) {
        if (negativeButton != null) {
            negativeButton.setVisibility(View.VISIBLE);
        }

        return this;
    }

    public AwesomeSuccessDialog showDoneButton(boolean show) {
        if (doneButton != null) {
            doneButton.setVisibility(View.VISIBLE);
        }

        return this;
    }

    public AwesomeSuccessDialog setPositiveButtonbackgroundColor(int buttonBackground) {
        if (positiveButton != null) {
            positiveButton.getBackground().setColorFilter(ContextCompat.getColor(getContext(), buttonBackground), PorterDuff.Mode.SRC_IN);
        }

        return this;
    }

    public AwesomeSuccessDialog setPositiveButtonTextColor(int textColor) {
        if (positiveButton != null) {
            positiveButton.setTextColor(ContextCompat.getColor(getContext(), textColor));
        }

        return this;
    }

    public AwesomeSuccessDialog setPositiveButtonText(String text) {
        if (positiveButton != null) {
            positiveButton.setText(text);
            positiveButton.setVisibility(View.VISIBLE);
        }

        return this;
    }

    public AwesomeSuccessDialog setNegativeButtonbackgroundColor(int buttonBackground) {
        if (negativeButton != null) {
            negativeButton.getBackground().setColorFilter(ContextCompat.getColor(getContext(), buttonBackground), PorterDuff.Mode.SRC_IN);
        }

        return this;
    }

    public AwesomeSuccessDialog setNegativeButtonText(String text) {
        if (negativeButton != null) {
            negativeButton.setText(text);
            negativeButton.setVisibility(View.VISIBLE);
        }

        return this;
    }

    public AwesomeSuccessDialog setNegativeButtonTextColor(int textColor) {
        if (negativeButton != null) {
            negativeButton.setTextColor(ContextCompat.getColor(getContext(), textColor));
        }

        return this;
    }

    public AwesomeSuccessDialog setDoneButtonbackgroundColor(int buttonBackground) {
        if (doneButton != null) {
            doneButton.getBackground().setColorFilter(ContextCompat.getColor(getContext(), buttonBackground), PorterDuff.Mode.SRC_IN);
        }

        return this;
    }

    public AwesomeSuccessDialog setDoneButtonText(String text) {
        if (doneButton != null) {
            doneButton.setText(text);
            doneButton.setVisibility(View.VISIBLE);
        }

        return this;
    }

    public AwesomeSuccessDialog setDoneButtonTextColor(int textColor) {
        if (doneButton != null) {
            doneButton.setTextColor(ContextCompat.getColor(getContext(), textColor));
        }

        return this;
    }


    @Override
    protected int getLayout() {
        return R.layout.dialog_success;
    }
}

File: awesomedialoglib/src/main/java/com/awesomedialog/blennersilva/awesomedialoglibrary/AwesomeSuccessDialog.kt
package com.awesomedialog.blennersilva.awesomedialoglibrary

import android.content.Context
import android.graphics.PorterDuff
import android.support.v4.content.ContextCompat
import android.view.View
import android.widget.Button
import android.widget.RelativeLayout

import com.awesomedialog.blennersilva.awesomedialoglibrary.interfaces.Closure

/**
 * Created by blennersilva on 23/08/17.
 */

class AwesomeSuccessDialog(context: Context) : AwesomeDialogBuilder<AwesomeSuccessDialog>(context) {

    private var positiveButton: Button? = findView(R.id.btDialogYes)
    private var negativeButton: Button? = findView(R.id.btDialogNo)
    private var doneButton: Button? = findView(R.id.btDialogDone)
    private var dialogBody: RelativeLayout = findView(R.id.dialog_body)


    override val layout: Int
        get() = R.layout.dialog_success

    init {
        setColoredCircle(R.color.dialogSuccessBackgroundColor)
        setDialogIconAndColor(R.drawable.ic_success, R.color.white)
        setNegativeButtonbackgroundColor(R.color.dialogSuccessBackgroundColor)
        setPositiveButtonbackgroundColor(R.color.dialogSuccessBackgroundColor)
        setDoneButtonbackgroundColor(R.color.dialogSuccessBackgroundColor)
    }

    fun setDialogBodyBackgroundColor(color: Int): AwesomeSuccessDialog {
        dialogBody.background.setColorFilter(ContextCompat.getColor(context!!, color), PorterDuff.Mode.SRC_IN)
        return this
    }

    fun setPositiveButtonClick(selectedYes: Closure?): AwesomeSuccessDialog {
        positiveButton!!.setOnClickListener {
            selectedYes?.exec()
            hide()
        }
        return this
    }

    fun setNegativeButtonClick(selectedNo: Closure?): AwesomeSuccessDialog {
        negativeButton!!.setOnClickListener {
            selectedNo?.exec()
            hide()
        }
        return this
    }

    fun setDoneButtonClick(selectedDone: Closure?): AwesomeSuccessDialog {
        doneButton!!.setOnClickListener {
            selectedDone?.exec()
            hide()
        }
        return this
    }

    fun showPositiveButton(show: Boolean): AwesomeSuccessDialog {
        if (positiveButton != null) {
            positiveButton!!.visibility = View.VISIBLE
        }
        return this
    }

    fun showNegativeButton(show: Boolean): AwesomeSuccessDialog {
        if (negativeButton != null) {
            negativeButton!!.visibility = View.VISIBLE
        }
        return this
    }

    fun showDoneButton(show: Boolean): AwesomeSuccessDialog {
        if (doneButton != null) {
            doneButton!!.visibility = View.VISIBLE
        }
        return this
    }

    fun setPositiveButtonbackgroundColor(buttonBackground: Int): AwesomeSuccessDialog {
        if (positiveButton != null) {
            positiveButton!!.background.setColorFilter(ContextCompat.getColor(context!!, buttonBackground), PorterDuff.Mode.SRC_IN)
        }
        return this
    }

    fun setPositiveButtonTextColor(textColor: Int): AwesomeSuccessDialog {
        if (positiveButton != null) {
            positiveButton!!.setTextColor(ContextCompat.getColor(context!!, textColor))
        }
        return this
    }

    fun setPositiveButtonText(text: String): AwesomeSuccessDialog {
        if (positiveButton != null) {
            positiveButton!!.text = text
            positiveButton!!.visibility = View.VISIBLE
        }
        return this
    }

    fun setNegativeButtonbackgroundColor(buttonBackground: Int): AwesomeSuccessDialog {
        if (negativeButton != null) {
            negativeButton!!.background.setColorFilter(ContextCompat.getColor(context!!, buttonBackground), PorterDuff.Mode.SRC_IN)
        }
        return this
    }

    fun setNegativeButtonText(text: String): AwesomeSuccessDialog {
        if (negativeButton != null) {
            negativeButton!!.text = text
            negativeButton!!.visibility = View.VISIBLE
        }
        return this
    }

    fun setNegativeButtonTextColor(textColor: Int): AwesomeSuccessDialog {
        if (negativeButton != null) {
            negativeButton!!.setTextColor(ContextCompat.getColor(context!!, textColor))
        }
        return this
    }

    fun setDoneButtonbackgroundColor(buttonBackground: Int): AwesomeSuccessDialog {
        if (doneButton != null) {
            doneButton!!.background.setColorFilter(ContextCompat.getColor(context!!, buttonBackground), PorterDuff.Mode.SRC_IN)
        }
        return this
    }

    fun setDoneButtonText(text: String): AwesomeSuccessDialog {
        if (doneButton != null) {
            doneButton!!.text = text
            doneButton!!.visibility = View.VISIBLE
        }
        return this
    }

    fun setDoneButtonTextColor(textColor: Int): AwesomeSuccessDialog {
        if (doneButton != null) {
            doneButton!!.setTextColor(ContextCompat.getColor(context!!, textColor))
        }
        return this
    }
}
